### PR TITLE
Add missing text in battle options dialog

### LINKED ISSUE #123: Add missing text in battle options dialog
During battle go to battle options dialog (left bottom button with computer icon) and notice this:
image

There is no any information regarding which option is for. Here is a screenshot from the original game:
image

As you could see here is some text. We don't have all options implemented but at least we must give some text for options we have like in original game.

### PR DESCRIPTION
**Pull request by MariaSopkova**
add new class "ButtonWithText". It can be used in menu dialogs.
Fixes #123

### CODE REVIEW

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 1]
```diff
@@ -42,6 +42,8 @@ namespace Battle
 {
     void GetSummaryParams(int res1, int res2, const HeroBase &, u32 exp, int &, std::string &);
     void SpeedRedraw(const Point &);
+    void doOnButtonClicked(ButtonWithText& button, Display & display, Cursor& cursor);
+    void initButtonState(bool state, ButtonWithText& button);
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp]: Please swap input parameters to be consistent.

Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp]: Let's call this functions as `OnButtonClicked` and `InitButtonState`.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 2]
```diff
@@ -54,7 +56,35 @@ void Battle::SpeedRedraw(const Point & dst)
...
 
     sprite.Blit(dst);
-    text.Blit(dst.x + (sprite.w() - text.w()) / 2, dst.y + sprite.h() - 15);
+    text.Blit(dst.x + (sprite.w() - text.w()) / 2, dst.y + sprite.h() + 3);
+
+    Text speedTitle("Speed", Font::SMALL);
+    speedTitle.Blit(dst.x + (sprite.w() - speedTitle.w()) / 2, dst.y - 13);
+}
+
+void Battle::doOnButtonClicked(ButtonWithText& button, Display & display, Cursor& cursor) {
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp]: Please use 4 space indentation and `{` on new line for function only.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 3]
```diff
@@ -54,7 +56,35 @@ void Battle::SpeedRedraw(const Point & dst)
...
+  }
+  button.Draw();
+  cursor.Show();
+  display.Flip();
+}
+
+void Battle::initButtonState(bool state, ButtonWithText& button) {
+  if (state) {
+    button.Press();
+    button.ResetButtomText("On");
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp]: I think we need to add translation support here. As I remember it's something like `_("On")`.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 4]
```diff
@@ -78,22 +108,22 @@ void Battle::DialogBattleSettings(void)
...
-
     Button btn_ok(pos_rt.x + 113, pos_rt.y + 252, (conf.ExtGameEvilInterface() ? ICN::CSPANBTE : ICN::CSPANBTN), 0, 1);
-
-    Rect   opt_speed(pos_rt.x + 36, pos_rt.y + 47, AGG::GetICN(ICN::CSPANEL, 0).w(), AGG::GetICN(ICN::CSPANEL, 0).h());
-    Button opt_grid(pos_rt.x + 36, pos_rt.y + 157, ICN::CSPANEL, 8, 9);
-    Button opt_shadow_movement(pos_rt.x + 128, pos_rt.y + 157, ICN::CSPANEL, 10, 11);
-    Button opt_shadow_cursor(pos_rt.x + 220, pos_rt.y + 157, ICN::CSPANEL, 12, 13);
+    SpriteBack speedButtom(Rect(pos_rt.x + 36, pos_rt.y + 47 + AGG::GetICN(ICN::CSPANEL, 0).h(),
+                          AGG::GetICN(ICN::CSPANEL, 0).h(), Text::height("speed", Font::SMALL) + 3));
+    Rect   opt_speed(pos_rt.x + 36, pos_rt.y + 47,
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp]: use `const` and a single space as well call this variable in CamelCase style as `OptSpeed`.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 5]
```diff
@@ -103,50 +133,34 @@ void Battle::DialogBattleSettings(void)
...
-	    opt_shadow_cursor.isPressed() ? opt_shadow_cursor.Release() : opt_shadow_cursor.Press();
-	    opt_shadow_cursor.Draw();
-	    cursor.Show();
-	    display.Flip();
-	}
-
-        // exit
-	if(Game::HotKeyPressEvent(Game::EVENT_DEFAULT_EXIT) || le.MouseClickLeft(btn_ok)) break;
+	    if(le.MouseClickLeft(opt_speed))
+	    {
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp]: Put `{` at the same line as if.

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 6]
```diff
@@ -103,50 +133,34 @@ void Battle::DialogBattleSettings(void)
...
-	    cursor.Hide();
-	    opt_shadow_cursor.isPressed() ? opt_shadow_cursor.Release() : opt_shadow_cursor.Press();
-	    opt_shadow_cursor.Draw();
-	    cursor.Show();
-	    display.Flip();
-	}
-
-        // exit
-	if(Game::HotKeyPressEvent(Game::EVENT_DEFAULT_EXIT) || le.MouseClickLeft(btn_ok)) break;
+	    if(le.MouseClickLeft(opt_speed))
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp]: A missing whitespace after word `if`

File: src/fheroes2/battle/battle_dialogs.cpp [hunk 7]
```diff
@@ -54,7 +56,35 @@ void Battle::SpeedRedraw(const Point & dst)
     const Sprite & sprite = AGG::GetICN(ICN::CSPANEL, (speed < 3 ? 0 : (speed < 7 ? 1 : 2)));
 
     sprite.Blit(dst);
-    text.Blit(dst.x + (sprite.w() - text.w()) / 2, dst.y + sprite.h() - 15);
+    text.Blit(dst.x + (sprite.w() - text.w()) / 2, dst.y + sprite.h() + 3);
+
+    Text speedTitle("Speed", Font::SMALL);
```
Comment (ihhub) [inline src/fheroes2/battle/battle_dialogs.cpp]: Use `const`.

File: src/fheroes2/gui/button.h [hunk 8]
```diff
@@ -58,6 +59,26 @@ class Button : public Rect
     u32		flags;
 };
 
+class ButtonWithText : public Button {
```
Comment (ihhub) [inline src/fheroes2/gui/button.h]: From OOD the name is really weird (I'm sorry if it's rude). I'm thinking if name `LabeledButton` is better?

**Comment by MariaSopkova** [comment]
@ihhub fixed your remarks

File: fheroes2.key [hunk 9]
```diff
@@ -103,4 +103,4 @@
 #
 #
 # 5. how to use this (example):
-default exit = 27
+# default exit = 27
```
Comment (ihhub) [inline fheroes2.key]: @MariaSopkova could you please explain why do we need this change? I'll fully review all changes on my weekends. My apologies for the delay.

Comment (MariaSopkova) [reply fheroes2.key]: Revert.

**Comment by ihhub** [comment]
Hi @MariaSopkova your changes are correct in relation to fixing issue. I'm thinking how to make the code more generic. Give me please some time to do it :)

**Comment by ihhub** [comment]
@MariaSopkova thank you very much for your continuous support of the project!

**Pull request merged**